### PR TITLE
Use consistent include paths when buiding in an external repository

### PR DESCRIPTION
**Pull request by agotsis**
Currently, the include paths passed to the jsonnet compiler are not all adjusted for when used in an external repository - only explicitly included imports.

This results in inconsistent include behavior when rules from rules_jsonnet are depended upon in an external repository.

Adding the [workspace_root](https://bazel.build/rules/lib/builtins/Label#workspace_root) as appropriate makes this behavior consistent between builds in a local workspace and in an external one.

All existing tests pass `bazel build @examples//...` with bazel 6.4.0.

### CODE REVIEW

File: jsonnet/jsonnet.bzl [hunk 1]
```diff
@@ -446,7 +461,9 @@ _jsonnet_common_attrs = {
         allow_files = True,
     ),
     "imports": attr.string_list(
-        doc = "List of import `-J` flags to be passed to the `jsonnet` compiler.",
+        doc = """List of import `-J` flags to be passed to the `jsonnet` compiler.
+
+The workspace_root, the genfiles_dir, and bin_dir are always included.""",
```
Comment (agotsis) [inline jsonnet/jsonnet.bzl:466]: This was already true. The generated documentation is not updated due to #177.

**Comment by agotsis** [comment]
The CI build is failing on master already - see #173.

**Comment by EdSchouten** [comment]
Thanks for the PR, @agotsis! I just merged a somewhat different change in #182 that should also address this issue. The reason I deviated from your change is because I think that we should simply make sure that the `imports` in the provider contains the correct set of paths to use. This also allows us to get rid of the genfiles/bindir logic.

**Comment by agotsis** [comment]
Thank you very much @EdSchouten - I have tested your changes in a large repo with complex dependencies on jsonnet and found that the issue that motivated me to open the PR has been fixed <3

**Pull request closed without merging**